portfolio/src/main/java/com/google/sps/servlets/PatentDataServlet.java

```diff
@@ -0,0 +1,48 @@
+package com.google.sps.servlets;
+
+import com.google.sps.models.PatentFiling;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Returns Patent data as a JSON array, e.g. [{"date": "2000-01-01", "lat": 38.40043, "lng": -122.23145}] */
+@WebServlet("/patent-data")
+public class PatentDataServlet extends HttpServlet {
+
+  private Collection<PatentFiling> patentFilings;
+
+  @Override
+  public void init() {
+    patentFilings = new ArrayList<>();
+
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream("/WEB-INF/canadian_patents.csv"));
+    // Go to next line to skip over column headings
+    scanner.nextLine();
+
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine();
+      String[] cells = line.split(",");
+
+      String date = cells[4];
```

Comment:
4 would be good to encapsulate in a variable 'int dateIndex = 4'

Reply:
5 and 6 would also be good to declare in an enum.

Reply:
Got it. Colette mentioned something about using an enum on my last PR as well. I haven't used them much in Java before but I did some more digging to understand how and when to use them and it makes sense now. Will try to use them more when I think I should moving forward.